contracts/staking/aave/GoodAaveStaking.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+import "../SimpleStaking.sol";
+import "../../Interfaces.sol";
+import "../../utils/DataTypes.sol";
+
+/**
+ * @title Staking contract that donates earned interest to the DAO
+ * allowing stakers to deposit Token
+ * or withdraw their stake in Token
+ * the contracts buy cToken and can transfer the daily interest to the  DAO
+ */
+contract GoodAaveStaking is SimpleStaking {
+	// Address of the TOKEN/USD oracle from chainlink
+	address public tokenUsdOracle;
+
+	//LendingPool of aave
+	ILendingPool public lendingPool;
+
+	/**
+	 * @param _token Token to swap DEFI token
+	 * @param _lendingPool LendingPool address
+	 * @param _ns Address of the NameService
+	 * @param _tokenName Name of the staking token which will be provided to staker for their staking share
+	 * @param _tokenSymbol Symbol of the staking token which will be provided to staker for their staking share
+	 * @param _tokenSymbol Determines blocks to pass for 1x Multiplier
+	 * @param _tokenUsdOracle address of the TOKEN/USD oracle
+	 * @param _collectInterestGasCost Gas cost for the collect interest of this staking contract
+	 */
+	function init(
+		address _token,
+		address _lendingPool,
+		NameService _ns,
+		string memory _tokenName,
+		string memory _tokenSymbol,
+		uint64 _maxRewardThreshold,
+		address _tokenUsdOracle,
+		uint32 _collectInterestGasCost
+	) public {
+		lendingPool = ILendingPool(_lendingPool);
+		DataTypes.ReserveData memory reserve = lendingPool.getReserveData(
+			_token
+		);
+		initialize(
+			_token,
+			reserve.aTokenAddress,
+			_ns,
+			_tokenName,
+			_tokenSymbol,
+			_maxRewardThreshold,
+			_collectInterestGasCost
+		);
+		//above  initialize going  to revert on second call, so this is safe
+		tokenUsdOracle = _tokenUsdOracle;
+	}
+
+	/**
+	 * @dev stake some Token
+	 * @param _amount of Token to stake
+	 */
+	function mintInterestToken(uint256 _amount) internal override {
+		lendingPool.deposit(address(token), _amount, address(this), 0);
+	}
+
+	/**
+	 * @dev redeem Token from aave
+	 * @param _amount of token to redeem in Token
+	 */
+	function redeem(uint256 _amount) internal override {
+		lendingPool.withdraw(address(token), _amount, address(this));
+	}
+
+	/**
+	 * @dev Function to redeem aToken for DAI, so reserve knows how to handle it. (reserve can handle dai or cdai)
+	 * @dev _amount of token in iToken
+	 * @return return address of the DAI and amount of the DAI
+	 */
+	function redeemUnderlyingToDAI(uint256 _amount)
+		internal
+		override
+		returns (address, uint256)
+	{
+		lendingPool.withdraw(address(token), _amount, address(this));
```

Comment:
maybe we can refactor redeemunderlying to send tokens directly to fundmanager, saving the transfer in simplestaking

Reply:
Yeah except if it's not cDAI we can directly send it to reserve I think